cnxauthoring/storage/postgresql.py

```diff
@@ -328,11 +329,20 @@ def update(self, item_or_items):
 
     def persist(self):
         """Persist/commit the changes."""
-        self.conn.commit()
+        # A try/except statement had been added to handle
+        # situations were the database connection is lost but not caught before
+        # a call to this method (see unittest test_functional.
+        # FunctionalTests.test_storage_unavailable_error). Would suggest
+        # moving all error catching inside this class in the future.
+        try:
+            self.conn.commit()
+        except self.Error:
+            raise httpexceptions.HTTPServiceUnavailable
```

Comment:
I forgot what database error @pumazi had when he created #167.  Did you check whether your test returns 200 OK without your fix?

I think the original issue has to do with code like this in views.py 

```
try:
    storage.persist()
except storage.Error:
    storage.abort()
```

If persist fails, abort is called and returns 200 OK eventually.  So you think you've saved your content but actually it's not saved in the database.

If @pumazi remembers what the database error was and it's not too hard to actually set it up, we can try to reproduce it in the tests.  Otherwise I suggest we just fake (mock) the database failure to test the code.